nodejs/week2/app.js

```diff
@@ -0,0 +1,93 @@
+import express from "express";
+import fs from "fs";
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+//Support parsing JSON requests
+app.use(express.json());
+
+let documents;
+try {
+  const data = fs.readFileSync("documents.json");
+  documents = JSON.parse(data);
+} catch (error) {
+  console.error("Error reading documents.json:", error);
+  documents = [];
+}
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+//Get /search
+app.get("/search", (req, res) => {
+  const query = req.query.q;
+
+  //If no query parameter is provided, return all documents
+  if (!query) {
+    return res.json(documents);
+  }
+
+  //Filter documents based on query parameter q
+  const result = documents.filter((doc) =>
+    Object.values(doc).some((value) =>
+      String(value).toLowerCase().includes(query.toLowerCase())
+    )
+  );
+
+  res.json(result);
+});
+
+//GET /documents/:id
+app.get("/documents/:id", (req, res) => {
+  const id = parseInt(req.params.id);
+  const document = documents.find((doc) => doc.id === id);
+
+  if (!document) {
+    return res.status(404).send("Document not found");
+  }
```

Comment:
Try to read about error handling here:
https://expressjs.com/en/guide/error-handling.html

you can put all your error handling into a middleware which will make your code more robust